lib/tty/file.rb

```diff
@@ -288,16 +288,26 @@ def create_file(relative_path, *args, context: nil, force: false, skip: false,
     # @api public
     def copy_file(source_path, *args, context: nil, force: false, skip: false,
                   verbose: true, color: :green, noop: false, preserve: nil, &block)
+
       source_path = source_path.to_s
+      unless ::File.file? source_path
+        log_status(:error, source_path, verbose: verbose, color: :red)
+        return
+      end
+
       dest_path = (args.first || source_path).to_s.sub(/\.erb$/, "")
 
+      if ::File.directory? dest_path
+        dest_path = ::File.join(dest_path, ::File.basename(source_path))
+      end
```

Comment:
Good catch! However, I think we shouldn't change the destination path. If someone wants to copy a file to an existing directory, we should allow for it and the file collision will handle the situation when the file already exists. For example,

```ruby
TTY::File.copy_file("foo/bar.rb", "baz/")
# => baz/bar.rb
```

Also, any changes like this require a test case.